providers/cncf/kubernetes/tests/unit/cncf/kubernetes/decorators/test_kubernetes.py

```diff
@@ -128,3 +130,31 @@ def f(arg1, arg2, kwarg1=None, kwarg2=None):
         # Second container is xcom image
         assert containers[1].image == XCOM_IMAGE
         assert containers[1].volume_mounts[0].mount_path == "/airflow/xcom"
+
+    def test_template_fields_order(self):
+        """Test that template_fields includes deterministic order."""
+        with self.dag:
+
+            @task.kubernetes(image="python:3.10-slim-buster")
+            def f():
+                return "test"
+
+            task_instance = f()
+
+        # Get the actual operator instance
+        operator = task_instance.operator
```

Comment:
That wasn't an task_instance (those only exist at runtime) -- it was actually an XComArg object, but regardless:

```suggestion
            operator = f().operator
```